c_src/sqlite3_nif.c

```diff
@@ -16,13 +16,14 @@
 #define MAX_PATHNAME    512
 
 static ErlNifResourceType* connection_type = NULL;
-static ErlNifResourceType* statement_type  = NULL;
+static ErlNifResourceType* statement_type = NULL;
 static sqlite3_mem_methods default_alloc_methods = {0};
 
 typedef struct connection
 {
     sqlite3* db;
     ErlNifMutex* mutex;
+    ErlNifPid update_hook_pid;
 } connection_t;
```

Comment:
I wonder if we could capture the `App.Repo` atom and store it with the connection and pass that in the callback instead of the raw db name. This would probably change how the sqlite database is opened.